icx-proxy/src/main.rs

```diff
@@ -131,7 +183,7 @@ async fn forward_request(
         builder = builder.header(&name, value);
     }
 
-    if let Some(streaming_strategy) = http_response.streaming_strategy {
+    let (body, response) = if let Some(streaming_strategy) = http_response.streaming_strategy {
```

Comment:
I think we can use the same logic for the `body` in the streaming and non-streaming case, because the logging code is only going to display the first few bytes anyway:

```suggestion
    let body = if logger.is_trace_enabled() {
        Some(http_response.body.clone())
    } else {
        None
    };
    let response = if let Some(streaming_strategy) = http_response.streaming_strategy {
```

Reply:
Right, but I'll change the code a bit to be more explicit here.

Reply:
or something like
```
let (first_part_of_body, total_body_length) = if logger.is_trace_enabled() {
    (Some(body[..usize::min(100, body.len()), Some(body.len()))
} else {
    (None, None)
}
```